internal/api/runme/runner/v1/runner.proto

```diff
@@ -0,0 +1,122 @@
+syntax = "proto3";
+
+package runme.runner.v1;
+
+import "google/protobuf/wrappers.proto";
+import "google/protobuf/duration.proto";
+
+option go_package = "github.com/stateful/runme/internal/gen/proto/go/runme/runner/v1;runnerv1";
+
+message Session {
+    string id = 1;
+
+    // envs keeps track of session environment variables.
+    // They can be modified by executing programs which
+    // alter them through "export" and "unset" commands.
+    repeated string envs = 2;
+
+    // metadata is a map of client specific metadata.
+    map<string, string> metadata = 3;
+}
+
+message CreateSessionRequest {
+    // metadata is a map of client specific metadata.
+    map<string, string> metadata = 1;
+
+    // envs field provides an initial set of environment variables
+    // for a newly created session.
+    repeated string envs = 2;
+}
+
+message CreateSessionResponse {
+    Session session = 1;
+}
+
+message GetSessionRequest {
+    string id = 1;
+}
+
+message GetSessionResponse {
+    Session session = 1;
+}
+
+message ListSessionsRequest {}
+
+message ListSessionsResponse {
+    repeated Session sessions = 1;
+}
+
+message DeleteSessionRequest {
```

Comment:
From my tests, I noticed that sessions are not automatically removed when its creator disconnects.

I realize it may not be trivial to implement such a thing. However, in principle, we are leaking environment variables unnecessarily if a client disconnects unexpectedly. It may be desirable from a security standpoint to try and prevent that.

Reply:
Good remarks. I think it's hard to tell at this point whether we want the server or client to have more responsibility. Yes, we can implement automatic session deletion, but that would mean more session restarts which we would like to avoid until we introduce sessions persistency. There're many tradeoffs which will be easier to resolve after we integrate both components and star using them.